kafka_consumer_freshness_tracker/src/main/java/com/tesla/data/consumer/freshness/ConsumerFreshness.java

```diff
@@ -113,22 +120,92 @@ public static void main(String[] args) throws IOException, InterruptedException
   }
 
   private void setup(Map<String, Object> conf) {
-    this.burrow = new Burrow((Map<String, Object>) conf.get("burrow"));
+    setupWithBurrow(conf, new Burrow((Map<String, Object>) conf.get("burrow")));
+  }
+
+  @VisibleForTesting
+  void setupWithBurrow(Map<String, Object> conf, Burrow burrow) {
+    this.burrow = burrow;
     int workerThreadCount = (int) conf.getOrDefault("workerThreadCount", DEFAULT_WORKER_THREADS);
     this.availableWorkers = ((List<Map<String, Object>>) conf.get("clusters")).stream()
         .map(clusterConf -> {
+          // validate the cluster configuration
+          Optional<String> validationErrorMsg = validateClusterConf(clusterConf);
+          if (validationErrorMsg.isPresent()) {
+            String msg = String.format("configuration for cluster %s is invalid: %s",
+                    clusterConf.get("name"),
+                    validationErrorMsg.get()
+            );
+            if (strict) {
+              throw new RuntimeException(msg);
+            } else {
+              LOG.warn(msg);
+            }
+          }
           // allow each cluster to override the number of workers, if desired
           int numConsumers = (int) clusterConf.getOrDefault("numConsumers", DEFAULT_KAFKA_CONSUMER_COUNT);
           ArrayBlockingQueue<KafkaConsumer> queue = new ArrayBlockingQueue<>(numConsumers);
           for (int i = 0; i < numConsumers; i++) {
             queue.add(createConsumer(clusterConf));
           }
           return new AbstractMap.SimpleEntry<>((String) clusterConf.get("name"), queue);
-        }).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+        })
+        .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
 
     this.executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workerThreadCount));
   }
 
+  //
+  //
+  //
+  //
+  //
+  //
+
+  /**
+   * Validate an individual cluster's configuration. A cluster's configuration is valid if:
+   * a) Looking it up by name in Burrow returns a successful response (not 404)
+   * b) Bootstrap servers match those in burrow
+   *  i. in strict mode: The bootstrap server list in the config matches the list advertised by Burrow exactly
+   *  ii. in normal mode: All entries in the config's bootstrap server list are advertised by Burrow (Burrow could
+   * have extras which don't appear in the config).
+   *
+   * @param clusterConf the cluster configuration
+   * @return an optional string which will contain the validation failure message if validation fails or null if
```

Comment:
Optional should be empty, not return null

Reply:
fixed